### PR TITLE
feat: Add support for flexible column name in JsonStreamWriter

### PR DESCRIPTION
**Pull request by tracyz-g**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [X] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigquerystorage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [X] Ensure the tests and linter pass
- [X] Code coverage does not decrease (if any source code was changed)
- [X] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

**Comment by tracyz-g** [comment]
@yirutang please help review the change.

File: google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/BigQuerySchemaUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class BigQuerySchemaUtilTest extends TestCase {
+
+  @Test
+  public void testIsProtoCompatible() {
+    List<String> protoCompatibleNames = Arrays.asList("col_1", "name", "_0_");
+    List<String> protoIncompatibleNames = Arrays.asList("0_col", "()", "列");
```
Comment (yirutang) [inline google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/BigQuerySchemaUtilTest.java:33]: nit, add a "a-1"

**Review by yirutang — APPROVED** [review]
Cool, really glad to see this works!

File: google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/BigQuerySchemaUtilTest.java (hunk 1 above)
Comment (tracyz-g) [reply google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/BigQuerySchemaUtilTest.java:33]: added

**Comment by Neenu1995** [comment]
Error message for windows check failure: 
```
Error: Exception in thread "Thread-340" io.grpc.StatusRuntimeException: CANCELLED: Interrupted while waiting for quota.
	at com.google.cloud.bigquery.storage.v1beta2.StreamWriterV2.maybeWaitForInflightQuota(StreamWriterV2.java:295)
	at com.google.cloud.bigquery.storage.v1beta2.StreamWriterV2.appendInternal(StreamWriterV2.java:273)
	at com.google.cloud.bigquery.storage.v1beta2.StreamWriterV2.append(StreamWriterV2.java:235)
	at com.google.cloud.bigquery.storage.v1beta2.StreamWriterV2Test.sendTestMessage(StreamWriterV2Test.java:132)
	at com.google.cloud.bigquery.storage.v1beta2.StreamWriterV2Test.access$000(StreamWriterV2Test.java:54)
	at com.google.cloud.bigquery.storage.v1beta2.StreamWriterV2Test$2.run(StreamWriterV2Test.java:158)
	at java.lang.Thread.run(Thread.java:750)
Caused by: java.lang.InterruptedException
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.reportInterruptAfterWait(AbstractQueuedSynchronizer.java:2019)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2178)
	at com.google.cloud.bigquery.storage.v1beta2.StreamWriterV2.maybeWaitForInflightQuota(StreamWriterV2.java:285)
	... 6 more
```

**Pull request merged**